File: DSA/Satyajit043/20-07-17/prime.cpp
#include<stdio.h>
#include<math.h>
int main()
{
	int b,n,i,c=1;
	printf("Enter no.\n");
	scanf("%d",&n);
	n = abs(n);
	if(n==2)
		printf("Prime. Found in 0 Divisions");
	else if(n%2==0 || sqrt(n)==int(sqrt(n)))
		printf("Not Prime. Found in %d Divisions",c);
	else
	{	b = int(sqrt(n));
		for(i=3;i<=b;i+=2)
		{	c++;
			if(n%i==0)
			{	printf("Not Prime. Found in %d Divisions",c);
				return 0;
			}
		}printf("Prime. Found in %d Divisions",c);
		 return 0;
	}return 0;
}

File: JAVA/bin/Satya043/Progs/28-07-17/Prime.java
import java.util.Scanner;
import java.math.*;
class Prime
{
	public static void main(String []args)
	{
		int a,i,b,c=0;
		Scanner sc = new Scanner(System.in);
		System.out.println("Enter Number");
		a = sc.nextInt();
		b = (int)(Math.sqrt(a));
		for(i=2;i<=b;i++)
			if(a%i==0)
				c++;
		if(c == 0)
			System.out.println("\nPrime");
		else
			System.out.println("\nNot Prime");
	}
}